src/vellum/workflows/graph/tests/test_graph.py

```diff
@@ -583,3 +583,20 @@ class SingleNode(BaseNode):
     # THEN it shows the single node
     assert "SingleNode.default" in result
     assert "Graph:" in result
+
+
+def test_graph__from_node_with_empty_ports():
```

Comment:
this test is perfect - now update our implementation of `Graph` to get this test to pass